### PR TITLE
add diagnostic messages display with virtual text and pointer

### PR DESCRIPTION
**Pull request by greeschenko**
Hi, I like your project and use it in my main .vimrc 

I suggest adding optional diagnostic messages inline display

### CODE REVIEW

File: doc/lsp.txt [hunk 1]
```diff
@@ -334,6 +334,9 @@ showInlayHints		|Boolean| option.  Show inlay hints from the language
 			server.  By default this is set to false.  The inlay
 			hint text is displayed as a virtual text.  Needs Vim
 			version 9.0.0178 or later.
+showDiagWithVirtualText		|Boolean| option.  Show diagnostic messages text from the language
+			server with virtual text.  By default this is set to false.
+            Needs Vim version 9.0.1157 or later.
 showSignature		|Boolean| option.  In insert mode, automatically show
```
[the remark this answers is not in the document]
Comment (greeschenko) [reply doc/lsp.txt:340]: yes of course i will add it

**Comment by andlrc** [comment]
For me it seems like the diagnostic popup (The popup drawn by `lsp#diag#LspDiagsJump -> DisplayDiag` fails when applying this PR:

BROKEN: https://asciinema.org/a/f41fSMzEZVtR9Je2pBR5WvPih

File: autoload/lsp/diag.vim [hunk 2]
```diff
@@ -41,6 +46,15 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
...
+
+    if has('patch-9.0.1157') && opt.lspOptions.showDiagWithVirtualText
+        prop_add( lnum, 0, {
+            'bufnr': bnr,
+            'type': 'LspDiagVirtualText',
+            'text': '┌─ ' .. diag.message,
+            'text_align': 'above',
+            'text_padding_left': diag.range.start.character})
+    endif
   endfor
```
Comment (yegappan) [inline autoload/lsp/diag.vim:58]: Can you modify the DiagsHighlightDisable() function to remove the virtual text?

File: autoload/lsp/diag.vim [hunk 3]
```diff
@@ -41,6 +46,15 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
     signs->add({id: 0, buffer: bnr, group: 'LSPDiag',
 				lnum: lnum,
 				name: DiagSevToSignName(diag.severity)})
+
+    if has('patch-9.0.1157') && opt.lspOptions.showDiagWithVirtualText
+        prop_add( lnum, 0, {
+            'bufnr': bnr,
```
Comment (yegappan) [inline autoload/lsp/diag.vim:52]: Can you remove the single quotes from the key names?  It is not needed in Vim9 script.

**Review by yegappan — CHANGES_REQUESTED** [review]

File: autoload/lsp/diag.vim [hunk 4]
```diff
@@ -27,6 +27,11 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
   # Remove all the existing diagnostic signs
```
Comment (yegappan) [inline autoload/lsp/diag.vim:27]: I noticed that when loading multiple buffers, there are error messages about trying to add a text property to an unloaded buffer.  At the beginning of this function, you need to add

   bnr->bufload()

to fix these issues.

File: autoload/lsp/diag.vim (hunk 3 above)
Comment (andlrc) [reply autoload/lsp/diag.vim:52]: > > I don't understand why though, as these shouldn't affect each other?
> 
> Did you try with the latest plugin? A redraw call was added before displaying the popup.

I tried it out on d161db7, see https://asciinema.org/a/0VAnghdBa73OatXW2lCC7vnDV

Comment (greeschenko) [reply autoload/lsp/diag.vim]: fix added in https://github.com/yegappan/lsp/pull/187/commits/6a7464d2dfa68fa5e4525e876712bcbed5f0300b

File: autoload/lsp/diag.vim (hunk 4 above)
Comment (greeschenko) [reply autoload/lsp/diag.vim:28]: added in https://github.com/yegappan/lsp/pull/187/commits/6a7464d2dfa68fa5e4525e876712bcbed5f0300b

File: autoload/lsp/diag.vim (hunk 2 above)
Comment (greeschenko) [reply autoload/lsp/diag.vim:66]: in https://github.com/yegappan/lsp/pull/187/commits/fdcb549003e49a48f5846911f25afaa4b2bef42d connect virtual text display to the autoHighlightDiags option as result we have on / off all Diags highlight and virtual text with :LspDiagHighlightEnable() :LspDiagHighlightDisable()

File: autoload/lsp/diag.vim (hunk 3 above)
Comment (greeschenko) [reply autoload/lsp/diag.vim]: > > > I don't understand why though, as these shouldn't affect each other?
> > 
> > 
> > Did you try with the latest plugin? A redraw call was added before displaying the popup.
> 
> I tried it out on [d161db7](https://github.com/yegappan/lsp/commit/d161db7c5d8ed8e2a4d73fa31e67dc54221f4eaa), same problem, see https://asciinema.org/a/0VAnghdBa73OatXW2lCC7vnDV

I think there is an error in the virtual text function at the editor level.
 I noticed that the virtual text doesn't always work correctly with pop-up dialogs.

**Pull request merged**